generator/integration_tests/tests/golden_kitchen_sink_rest_logging_decorator_test.cc

```diff
@@ -0,0 +1,258 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "generator/integration_tests/golden/internal/golden_kitchen_sink_rest_logging_decorator.h"
+#include "google/cloud/internal/rest_context.h"
+#include "google/cloud/testing_util/scoped_log.h"
+#include "google/cloud/testing_util/status_matchers.h"
+#include "absl/memory/memory.h"
+#include "generator/integration_tests/tests/mock_golden_kitchen_sink_rest_stub.h"
+#include <gmock/gmock.h>
+#include <memory>
+
+namespace google {
+namespace cloud {
+namespace golden_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+namespace {
+
+using ::testing::Contains;
+using ::testing::HasSubstr;
+using ::testing::Return;
+
+class LoggingDecoratorRestTest : public ::testing::Test {
```

Comment:
This is another class where the fixture makes the code harder to read than just defining `mock_` and `log_` locally in each test.